File: easy/1232_check_if_it_is_a_straight_line/solution.cpp
#include <vector>

class Solution {
public:
  bool checkStraightLine(std::vector<std::vector<int>> &coordinates) {
    for (size_t i = 2; i < coordinates.size(); ++i) {
      if ((coordinates[i - 1][1] - coordinates[i - 2][1]) *
              (coordinates[i][0] - coordinates[i - 2][0]) !=
          (coordinates[i][1] - coordinates[i - 2][1]) *
              (coordinates[i - 1][0] - coordinates[i - 2][0]))
        return false;
    }

    return true;
  }
};

File: easy/1232_check_if_it_is_a_straight_line/solution.py
class Solution:
    def checkStraightLine(self, coordinates: list[list[int]]) -> bool:
        for i in range(2, len(coordinates)):
            if (coordinates[i - 1][1] - coordinates[i - 2][1]) * (
                coordinates[i][0] - coordinates[i - 2][0]
            ) != (coordinates[i][1] - coordinates[i - 2][1]) * (
                coordinates[i - 1][0] - coordinates[i - 2][0]
            ):
                return False

        return True
